phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryWithTableSampleIT.java

```diff
@@ -0,0 +1,261 @@
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        try {
+            prepareTableWithValues(conn, 100);
+            String query = "SELECT count(i1) FROM " + tableName +" tablesample (22) where i2>=3000 or i1<2 ";
+            ResultSet rs = conn.createStatement().executeQuery(query);
+            
+            assertTrue(rs.next());
+            assertEquals(14, rs.getInt(1));
+        } finally {
+            conn.close();
+        }
+    }
+    
+    @Test
+    public void testSingleQueryWithUnion() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        try {
+            prepareTableWithValues(conn, 100);
+            String query = "SELECT * FROM " + tableName +" tablesample (100) where i1<2 union all SELECT * FROM " + tableName +" tablesample (2) where i2<6000";
+            ResultSet rs = conn.createStatement().executeQuery(query);
+            
+            assertTrue(rs.next());
+            assertEquals(0, rs.getInt(1));
+            assertTrue(rs.next());
+            assertEquals(1, rs.getInt(1));
+            assertTrue(rs.next());
+            assertEquals(30, rs.getInt(1));
+            assertTrue(rs.next());
+            assertEquals(44, rs.getInt(1));
+        } finally {
+            conn.close();
+        }
+    }
+    
+    @Test
+    public void testSingleQueryWithSubQuery() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        try {
+            prepareTableWithValues(conn, 100);
+            String query = "SELECT count(*) FROM (SELECT * FROM " + tableName +" tablesample (50))";
+            ResultSet rs = conn.createStatement().executeQuery(query);
+            
+            assertTrue(rs.next());
+            assertEquals(50, rs.getInt(1));
+        } finally {
+            conn.close();
+        }
+    }
+    
+    @Test
+    public void testSingleQueryWithJoins() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TestUtil.TEST_PROPERTIES);
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        try {
+            prepareTableWithValues(conn, 100);
+            String query = "SELECT count(*) FROM " + tableName +" as A tablesample (45), "+joinedTableName+" as B tablesample (75) where A.i1=B.i1";
+            System.out.println(query);
+            ResultSet rs = conn.createStatement().executeQuery(query);
+
+            assertTrue(rs.next());
+            assertEquals(31, rs.getInt(1));
+        } finally {
+            conn.close();
+        }
+    }
+    
+    final private static void inspect(final ResultSet rs) throws SQLException{
+    	while(rs.next()){
+    		System.out.println(rs.getInt(1)+","+rs.getInt(2));
```

Comment:
Minor nit: we don't write to system.out for testing (though I understand the value while debugging). Perhaps simplest to just switch to logging instead.